chemprop/cli/train.py

```diff
@@ -620,7 +639,7 @@ def build_model(args, train_dset: MolGraphDataset | MulticomponentDataset) -> MP
         # if args.mpn_shared:
         #     mp_block = MulticomponentMessagePassing(mp_blocks[0], n_components, args.mpn_shared)
         # else:
-        d_xd = sum(dset.d_xd for dset in train_dset.datasets)
+        d_xd = train_dset.datasets[0].d_xd
```

Comment:
Yes, we only take the x_d from the first dataset (because this is a once per datapoint tensor regardless of how many components) so we only need to read the dimension of the x_d in the first dataset. https://github.com/chemprop/chemprop/blob/57030d47b5ff95e51b4bee6a526861e63f7bcd6c/chemprop/data/collate.py#L113